mlir/test/Transforms/memref-alloc-to-gpu-alloc.mlir

```diff
@@ -0,0 +1,23 @@
+// RUN: level-zero-opt --insert-gpu-alloc %s | FileCheck %s
+
+func.func @main() {
+  %c8 = arith.constant 8 : index
+  %c1 = arith.constant 1 : index
+  // CHECK: func.func @main()
+  %0 = memref.alloc() : memref<8xf32>
+  %1 = memref.alloc() : memref<8xf32>
+  %2 = memref.alloc() : memref<8xf32>
+  // CHECK: %memref = gpu.alloc  () : memref<8xf32>
+  // CHECK: %memref_0 = gpu.alloc  () : memref<8xf32>
+  // CHECK: %memref_1 = gpu.alloc  () : memref<8xf32>
```

Comment:
Please check there are actually loads from these gpu allocs inside gpu.launch

Reply:
what do you mean?

Reply:
Something like
```
// CHECK: %[[MEMREF0:.*]]= gpu.alloc  () : memref<8xf32>
// CHECK: gpu.launch
// CHECK: memref.load %[[MEMREF0]][...]
// CHECK: gpu.terminator
```

Reply:
@nbpatel please address this comment and then it is LGTM

Reply:
@lchang552

Reply:
This is revised. Please let me know if further changes are required. -Liangliang